server/app/mutations/grid_services/common.rb

```diff
@@ -84,6 +86,30 @@ def build_grid_service_secrets(existing_secrets)
       service_secrets
     end
 
+    def build_service_volumes()
+      service_volumes = []
+      self.volumes.each do |vol|
+        vol_spec = parse_volume(vol)
```

Comment:
Needs to handle the error case of `vol` not being a valid volume spec, like `"foo"`. Currently `parse_volume` may return `nil`, and then this explodes.